pkg/limayaml/defaults.go

```diff
@@ -90,14 +91,10 @@ func defaultContainerdArchives() []File {
 
 // FirstUsernetIndex gets the index of first usernet network under l.Network[]. Returns -1 if no usernet network found
 func FirstUsernetIndex(l *LimaYAML) int {
-	for i := range l.Networks {
-		nwName := l.Networks[i].Lima
-		isUsernet, _ := networks.Usernet(nwName)
-		if isUsernet {
-			return i
-		}
-	}
-	return -1
+	return slices.IndexFunc(l.Networks, func(network Network) bool {
+		isUsernet, _ := networks.Usernet(network.Lima)
+		return isUsernet
```

Comment:
Maybe outside the scope of this PR, but I was initially confused by the `Usernet` method name, until I realized that it just returns a boolean. So maybe `IsUsernet` would be a better name?

Or maybe it should be an accessor to the network mode instead, to be a more general API that would then use like this:

```suggestion
		mode, _ := networks.Mode(network.Lima)
		return mode == networks.ModeUsernet
```

This is the only location that calls `networks.Usernet`, so maybe we can ignore it until there are other use cases.

Reply:
I renamed to `networks.IsUsernet` as think it's the most suitable solution, at least for now.